cognite/gqlpygen/main.py

```diff
@@ -1,60 +1,165 @@
+        module_name = schema_module.stem
+        spec = importlib.util.spec_from_file_location(module_name, schema_module)
+        module = importlib.util.module_from_spec(spec)  # type:ignore[arg-type]
+        sys.modules[module_name] = module
+        spec.loader.exec_module(module)  # type:ignore[union-attr]
+    else:
+        module_name = settings.local.get("schema_module", default=schema_module.stem)
+        click.echo(f"Got module {schema_module}, trying to import it...")
+        module = importlib.import_module(module_name)
     click.echo("Import successful")
 
-    click.echo("Searching for a schema...")
-    for schema_name, instance in inspect.getmembers(module):
-        if isinstance(instance, Schema):
-            click.echo(f"Found schema {schema_name!r}")
-            break
+    if not _prefix:
+        click.echo("Searching for a schema...")
+        for schema_name, instance in inspect.getmembers(module):
+            if isinstance(instance, Schema):
+                click.echo(f"Found schema {schema_name!r}")
+                break
+        else:
+            click.echo("Failed to find schema, exiting..")
+            exit(1)
     else:
-        click.echo("Failed to find schema, exiting..")
-        exit(1)
+        schema_name = to_schema_name(prefix)
+        click.echo(f"Got schema name {schema_name}")
+        instance = getattr(module, schema_name)
 
-    output = Path.cwd() / "schema.graphql"
-    output.write_text(instance.as_str())
-    click.echo(f"Wrote file {output}")
+    graphql_schema.write_text(instance.as_str())
+    click.echo(f"Wrote file {graphql_schema}")
+
+
+@app.command("signin", help="Upload a GQL schema to CDF DM data model.")
+def signin(
+    cdf_cluster: str = typer.Option(_cdf_cluster or ..., help="CDF cluster name."),
+    tenant_id: str = typer.Option(_tenant_id or ..., help="AD tenant ID."),
+    client_id: str = typer.Option(_client_id or ..., help="AD client ID."),
+    client_secret: str = typer.Option(
+        _client_secret_hidden or None, prompt=True, hide_input=True, help="AD client secret."
+    ),
+    project: str = typer.Option(_project or ..., help="Name of CDF project."),
+):
+    if client_secret == _client_secret_hidden:
+        client_secret = _client_secret
+    if client_secret == _client_secret_none:
+        auth_option = "--device-code"
+        auth_option_hidden = None
+    else:
+        auth_option = f"--client-secret='{client_secret}'"
+        auth_option_hidden = f"--client-secret='{client_secret[:3]}{'*' * (len(client_secret) - 3)}'"
+
+    command = [
+        "cdf",
+        "signin",
+        f"'{project}'",
+        f"--cluster='{cdf_cluster}'",
+        f"--tenant='{tenant_id}'",
+        f"--client-id='{client_id}'",
+        auth_option,
+    ]
+    msg = f"Executing:\n{' '.join(command)}"
+    if auth_option_hidden:
+        msg = msg.replace(auth_option, auth_option_hidden)
+    typer.echo(f"Executing:\n{msg}")
+    subprocess.call(command)
```

Comment:
```suggestion
    subprocess.run(command, shell=True)
```

I tested this on Windows and needed to do the following to get it to run.
* .call seems to be an old API kept for compatibility, thus I think .run should be used https://docs.python.org/3/library/subprocess.html#older-high-level-api
* Why shell=True is necessary for Windows do not know.